keeper/component/dispatcher/dispatcher.go

```diff
@@ -124,32 +127,30 @@ func (d *Dispatcher) DispatchPayload(
 	transferAttr *types.TransferAttributes,
 	payload *core.Payload,
 ) error {
-	if err := d.validatePayload(payload); err != nil {
+	if err := d.ValidatePayload(payload); err != nil {
+		// TODO: look at how it is displayed
 		return core.ErrValidation.Wrap(err.Error())
 	}
 
 	if err := d.dispatchActions(ctx, transferAttr, payload.PreActions); err != nil {
-		return fmt.Errorf("actions dispatch failed: %w", err)
+		return errors.Wrap(err, "actions dispatch failed")
 	}
 
 	if err := d.dispatchForwarding(ctx, transferAttr, payload.Forwarding); err != nil {
-		return fmt.Errorf("forwarding dispatch failed: %w", err)
+		return errors.Wrap(err, "forwarding dispatch failed")
 	}
 
 	if err := d.UpdateStats(ctx, transferAttr, payload.Forwarding); err != nil {
-		d.logger.Error("Error updating Orbiter statistics", "error", err.Error())
+		// NOTE: we don't want to interrupt a dispatch because stats are not updated.
```

Comment:
I think it's clearer like this
```suggestion
		// NOTE: we don't want to interrupt a dispatch in case the stats are not updated.
```